### DISCUSSION TITLE
If I want to compile osqp,eigen and osqp-eigen with visual studio 2015, which version of the latter three should I use, and which cmake version?

### DISCUSSION THREAD

**Discussion by DRETQ** [osqp/osqp#655] [Eigen Interface (Community)]
If I want to compile osqp,eigen and osqp-eigen with visual studio 2015, which version of the latter three should I use, and which cmake version?

**Comment by traversaro** [comment]
Hello @DRETQ, I am one of the osqp-eigen mantainers. I am not sure has exactly the answer to your question. However, looking a bit at the CI history, it seems that the latest version of osqp-eigen that run under VS2015 in CI was osqp-eigen v0.5.2, see https://github.com/robotology/osqp-eigen/blob/v0.5.2/appveyor.yml#L9 . In that CI, we were compiling with Eigen `3.3-beta2` (but perhaps also Eigen 3.3.* may work) while for osqp in the CI we were not using any tag, but the release notes mentioned that osqp-eigen v0.5.2 should be compatible with osqp 0.6.0 : https://github.com/robotology/osqp-eigen/releases/tag/v0.5.2 .

Having said that, it is still possible you find some unexpected error when using such old and unsupported compiler.

**Comment by DRETQ** [comment]
@traversaro Thanks for your reply, i'll try to see if it works with these appropriate versions. The purpose of using VS 2015 is mainly for compatibility with matlab 2015.

**Comment by DRETQ** [comment]
@traversaro I made an attempt and after changing versions, I did have problems compiling in vs2015 despite the installation and configuration steps being the same as the original.

**Reply by traversaro** [reply]
Unfortunatly that is perfectly possible, that is what I was referring with:

> Having said that, it is still possible you find some unexpected error when using such old and unsupported compiler.

One set of problems that you may encounter is that the version of VS2015 and/or the Windows SDK tested back in time was an earlier version w.r.t. to the one that you are installing.

> The purpose of using VS 2015 is mainly for compatibility with matlab 2015.

If you do not have any other constraint you can also give it a short with mingw, we never tested it but you could have better luck.

**Comment by DRETQ** [comment]
@traversaro Thanks for your reply, I'm not sure if it's a problem with vs2015 or osqp 0.6.0,eigen 3.3.9,osqp-eigen 0.5.2, I found that the name of the static link library in the old version of osqp-eigen (OsqpEigen) is different from the one in the newer version (OsqpEigend) and that the selection of the Cmake version is the same when the vs2015 compilation to choose x64.

**Reply by traversaro** [reply]
`OsqpEigen` vs `OsqpEigend` is not a matter of old or new, it depends on the `CMAKE_BUILD_TYPE` or build type, `d` is added in Debug mode, see https://github.com/robotology/osqp-eigen/blob/a752c5e6353cae6176578488a1fa641d3b9cf0b9/CMakeLists.txt#L29 .

**Comment by DRETQ** [comment]
@traversaro Do you know where to download the original osqp0.6.0, I'm not sure I've got it right!

**Reply by traversaro** [reply]
You can get it at https://github.com/osqp/osqp/releases/tag/v0.6.0 .

**Comment by DRETQ** [comment]
@traversaro Hello Professor, I have a question, according to the official website to install osqp-eigen, if I don't use mingw compile but use visual studio 14 2015, I change the command is like this, I choose the folder to put the source code D:\Qt\osqp-eigen-master, please ask this modification is correct, the last two steps are necessary ( make
make install), I found the problem in practice!
![1](https://github.com/osqp/osqp/assets/102904381/3816456a-f39a-40bd-ba51-f099a5e1f1c4)
![2](https://github.com/osqp/osqp/assets/102904381/a4e7cf4f-c923-494b-8e74-b726ad34676f)

**Reply by traversaro** [reply]
IF you use Visual Studio, you need to change `make` to `cmake --build .` and `make isntall` to `cmake --install .`.

**Comment by DRETQ** [comment]
@traversaro Thanks, I've solved the problem.

**Comment by DRETQ** [comment]
@traversaro I would like to ask if you have ever done calling C code containing osqp, eigen and osqp-eigen in a lower version of matlab/simulink, such as 2015b, using the s function builder module, such as the official example file MPCExample?

**Reply by traversaro** [reply]
> @traversaro I would like to ask if you have ever done calling C code containing osqp, eigen and osqp-eigen in a lower version of matlab/simulink, such as 2015b, using the s function builder module, such as the official example file MPCExample?

No, we used https://github.com/robotology/wb-toolbox to call osqp in Simulink, but I think the OSQP block was added after we stopped supporting vs2015
